pkg/usagestats/reporter.go

```diff
@@ -3,18 +3,34 @@
 package usagestats
 
 import (
+	"bytes"
 	"context"
+	"encoding/json"
 	"flag"
+	"fmt"
+	"io"
+	"math"
+	"net/http"
+	"time"
 
 	"github.com/go-kit/log"
 	"github.com/go-kit/log/level"
-	"github.com/grafana/dskit/services"
 	"github.com/pkg/errors"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
 	"github.com/thanos-io/thanos/pkg/objstore"
 
+	"github.com/grafana/dskit/services"
+
 	"github.com/grafana/mimir/pkg/storage/bucket"
 )
 
+const (
+	defaultReportCheckInterval = time.Minute
+	defaultReportSendInterval  = 4 * time.Hour
+	defaultStatsServerURL      = "https://stats.grafana.org/mimir-usage-report"
```

Comment:
Note to reviewers: doesn't exist yet. I'm building the support in Mimir following the Loki specs.